src/bridge/cockpitdisksamples.c

```diff
@@ -159,37 +161,34 @@ cockpit_disk_samples (CockpitSamples *samples)
   g_free (contents);
 }
 
-static gchar*
-read_file (const gchar *path)
+static FILE*
+open_file (int dirfd,
+           const char *name)
 {
-  g_autoptr(GError) error = NULL;
-  gchar *contents = NULL;
-  if (!g_file_get_contents (path, &contents, NULL, &error))
+  const int fd = openat (dirfd, name, O_RDONLY);
+  if (fd < 0)
     {
-      /* Do not log:
-       * - file does not exist: process died and can no longer be sampled
-       * - access denied: bridge is not privileged and can't sample all processes
-       * - sometimes ERROR_FAILED is returned with message: "No such process"
-       */
-      if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT) ||
-          g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_ACCES) ||
-          (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_FAILED) && g_strrstr(error->message, "No such process")))
-        {
-          return NULL;
-        }
+      if (errno != EACCES)
+        g_message ("error opening file descriptor: %s: %m", name);
+      return NULL;
+    }
 
-      g_warning ("Domain: %s, code: %d, message: %s", g_quark_to_string (error->domain), error->code, error->message);
+  FILE *fp = fdopen (fd, "r");
+  if (!fp)
+    {
+      if (errno != EACCES)
+        g_message ("error opening file: %s: %m", name);
```

Comment:
I think this can't fail except with `ENOMEM`.  You'd get the permission error only from the `openat()` call.

Reply:
Honestly, if this does fail, I'd `g_error()` to abort the program.  If you don't want to do that, then you need to `close (fd);`.